pkg/fanal/image/registry/azure/azure.go

```diff
@@ -24,13 +35,26 @@ func (r *Registry) CheckOptions(domain string, _ types.RegistryOptions) error {
 }
 
 func (r *Registry) GetCredential(ctx context.Context) (string, string, error) {
-	credStore, err := NewACRCredStore()
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
+	if err != nil {
+		return "", "", xerrors.Errorf("unable to generate acr credential error: %w", err)
+	}
+	aadToken, err := cred.GetToken(ctx, policy.TokenRequestOptions{Scopes: []string{scope}})
 	if err != nil {
-		return "", "", xerrors.Errorf("ACR credential error: %w", err)
+		return "", "", xerrors.Errorf("unable to get an access token: %w", err)
 	}
-	token, err := credStore.Get(ctx, r.domain)
+	rt, err := refreshToken(ctx, aadToken.Token, r.domain)
 	if err != nil {
-		return "", "", xerrors.Errorf("unable to get a token: %w", err)
+		return "", "", xerrors.Errorf("unable to refresh token: %w", err)
+	}
+	return "00000000-0000-0000-0000-000000000000", *rt.RefreshToken, err
```

Comment:
It returns a refresh token. Can it be used for registry authentication? In general, refresh tokens are used for renewing access tokens. I just want to make sure of it.

Reply:
trivy can't use `AAD token` for ACR authentication,  the `GetFromExchange()` method exchange `AAD token` with `ACR token` , see grant type is `access_token`.

AAD token will fail authentication for ACR. t is compatible with [prev. Implementation](https://github.com/aquasecurity/trivy/blob/main/pkg/fanal/image/registry/azure/auth.go#L96)

Reply:
Thanks for your explanation!